components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/CoreListener.java

```diff
@@ -0,0 +1,270 @@
+        this.subject = subject;
+        this.injectHandler = injectHandler;
+        this.natsProperties = natsProperties;
+    }
+
+    @Override public boolean createConnection() throws IOException, InterruptedException {
+        if (connection == null) {
+            connection = getNatsConnection();
+        }
+        return true;
+    }
+
+    public Connection getNatsConnection() throws IOException, InterruptedException {
+        String bufferSize = natsProperties.getProperty(NatsConstants.BUFFER_SIZE);
+        String turnOnAdvancedStats = natsProperties.getProperty(NatsConstants.TURN_ON_ADVANCED_STATS);
+        String traceConnection = natsProperties.getProperty(NatsConstants.TRACE_CONNECTION);
+        String tlsProtocol = validateParameter(natsProperties.getProperty(NatsConstants.TLS_PROTOCOL));
+        String tlsKeyStoreType = validateParameter(natsProperties.getProperty(NatsConstants.TLS_KEYSTORE_TYPE));
+        String tlsKeyStoreLocation = validateParameter(natsProperties.getProperty(NatsConstants.TLS_KEYSTORE_LOCATION));
+        String tlsKeyStorePassword = validateParameter(natsProperties.getProperty(NatsConstants.TLS_KEYSTORE_PASSWORD));
+        String tlsTrustStoreType = validateParameter(natsProperties.getProperty(NatsConstants.TLS_TRUSTSTORE_TYPE));
+        String tlsTrustStoreLocation = validateParameter(natsProperties.getProperty(NatsConstants.TLS_TRUSTSTORE_LOCATION));
+        String tlsTrustStorePassword = validateParameter(natsProperties.getProperty(NatsConstants.TLS_TRUSTSTORE_PASSWORD));
+        String tlsKeyManagerAlgorithm = validateParameter(natsProperties.getProperty(NatsConstants.TLS_KEY_MANAGER_ALGORITHM));
+        String tlsTrustManagerAlgorithm = validateParameter(natsProperties.getProperty(NatsConstants.TLS_TRUST_MANAGER_ALGORITHM));
+
+        Options.Builder builder = new Options.Builder(natsProperties);
+
+        if (StringUtils.isNotEmpty(bufferSize)) {
+            builder.bufferSize(Integer.parseInt(bufferSize));
+        }
+
+        if (Boolean.parseBoolean(turnOnAdvancedStats)) {
+            builder.turnOnAdvancedStats();
+        }
+
+        if (Boolean.parseBoolean(traceConnection)) {
+            builder.traceConnection();
+        }
+
+        if (StringUtils.isNotEmpty(
+                tlsProtocol + tlsTrustStoreType + tlsTrustStoreLocation + tlsTrustStorePassword + tlsKeyStoreType
+                        + tlsKeyStoreLocation + tlsKeyStorePassword + tlsKeyManagerAlgorithm
+                        + tlsTrustManagerAlgorithm)) {
+            SSLContext sslContext = createSSLContext(
+                    new TLSConnection(tlsProtocol, tlsTrustStoreType, tlsTrustStoreLocation, tlsTrustStorePassword,
+                            tlsKeyStoreType, tlsKeyStoreLocation, tlsKeyStorePassword, tlsKeyManagerAlgorithm,
+                            tlsTrustManagerAlgorithm));
+            if (sslContext != null) {
```

Comment:
Do we need this null check? If there was an issue creating SSLContext, IOException would be thrown and this code will not be executed.